app/controllers/species_list_controller.rb

```diff
@@ -390,6 +392,53 @@ def destroy_species_list # :norobots:
     end
   end
 
+  def add_remove_observations # :prefetch: :norobots:
+    pass_query_params
+    @id = params[:species_list].to_s
+    @query = find_query(:Observation)
+    if !@query
+      flash_error(:species_list_add_remove_no_query.t)
+      redirect_to(action: "list_species_lists")
+    end
+  end
+
+  def post_add_remove_observations # :prefetch: :norobots:
```

Comment:
I'd like to see this function broken up and I'm a bit concerned that Rubocop is reporting 17 new "offenses".